### ISSUE TITLE
Remove deprecated twisted.web.trp

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4299]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#4299|
|Type|task|
|Created|2010-02-23 00:50:57Z|
|Branch|https://github.com/twisted/twisted/tree/remove-trp-4299|

[twisted/web/trp.py](twisted/web/trp.py) was deprecated a year ago, let's remove it.

<details><summary>Searchable metadata</summary>

```
trac-id__4299 4299
type__task task
reporter__thijs thijs
priority__normal normal
milestone__ 
branch__branches_remove_trp_4299 branches/remove-trp-4299
branch_author__cyli cyli
status__closed closed
resolution__fixed fixed
component__web web
keywords__easy easy
time__1266886257000000 1266886257000000
changetime__1277919454000000 1277919454000000
version__None None
owner__ 
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra set owner to @cyli|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_2"></a>|@cyli commented|
|-|-|

(In [29459]) Branching to 'remove-trp-4299'

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_3"></a>|@cyli commented|
|-|-|

(In [29460]) Removed twisted/web/trp.py and any references to it.

refs [#4299](https://github.com/twisted/twisted/issues/4299)

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_4"></a>|@cyli commented|
|-|-|

(In [29461]) Copyright bump/removal of unused imports

refs [#4299](https://github.com/twisted/twisted/issues/4299)

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_5"></a>|@cyli removed owner|
|-|-|

http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/remove-trp-4299

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set owner to @cyli|
|-|-|

  1. `twisted.web.static.File.createPickleChild` should be removed as well, and _index.trp_ should be removed from `File.indexNames`.
  1. There's one more mention in the docs, in doc/web/howto/using-twistedweb.xhtml, _Of course, an equivalent ..._

Thanks

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_7"></a>|@cyli commented|
|-|-|

(In [29489]) Removed some more references to trp in twisted.web.static and twisted.web2.static, as well as twisted.web.test.test_static, and other files.

refs [#4299](https://github.com/twisted/twisted/issues/4299)

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)|@cyli removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun set owner to @cyli|
|-|-|

Sorry, there were _two_ references to trp in `doc/web/howto/using-twistedweb.xhtml`.  Now there's one. :)  Remove that and then merge.

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_10"></a>|@cyli set status to `closed`|
|-|-|

(In [29522]) Merge remove-trp-4299 : Removed twisted.web.trp

Author: cyli
Reviewer: exarkun
Fixes [#4299](https://github.com/twisted/twisted/issues/4299)

Removed deprecated twisted.web.trp and all the references to it in other files and tests.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
